<launch>
        <!-- 1. start the robot arm  and Moveit -->
        <include file="$(find xarm_driver)/launch/xarm_driver.launch" />
        <include file="$(find xarm_moveit_config)/launch/xarm_moveit_planning_execution.launch" />

        <!-- 2. start the Realsense415 camera -->
        <include file="$(find realsense2_camera)/launch/rs_rgbd.launch" />

        <!-- 3. publish TF between base_link and camera_link -->
        <include file="$(find vision_demo)/launch/publish_hand_eye_tf.launch" />

        <arg name="marker_size" default="4.5" />
        <arg name="max_new_marker_error" default="0.08" />
        <arg name="max_track_error" default="0.05" />

        <arg name="cam_image_topic" default="/camera/depth_registered/points" />
        <arg name="cam_info_topic" default="/camera/depth/camera_info" />
        <arg name="output_frame" default="/base_link" />

        <!-- 4. start the ar_track_alvar node -->
        <node name="ar_track_alvar" pkg="ar_track_alvar" type="individualMarkers" respawn="false" output="screen">
          <param name="marker_size"           type="double" value="$(arg marker_size)" />
          <param name="max_new_marker_error"  type="double" value="$(arg max_new_marker_error)" />
          <param name="max_track_error"       type="double" value="$(arg max_track_error)" />
          <param name="output_frame"          type="string" value="$(arg output_frame)" />
          <remap from="camera_image"  to="$(arg cam_image_topic)" />
          <remap from="camera_info"   to="$(arg cam_info_topic)" />
        </node>

        <!-- 5. start the Rviz -->
        <node name="rviz" pkg="rviz" type="rviz" args="-d $(find vision_demo)/rviz/ar_tag_pick.rviz" />
</launch>
